crates/launcher/src/entrypoint/native/mod.rs

```diff
@@ -0,0 +1,274 @@
+            use std::io::Write;
+
+            file.write_all(report.as_bytes())?;
+            file.flush()?;
+            let (_, path) = file.keep()?;
+            Ok(path)
+        }(report));
+        env::set_var("LUMINOL_PANIC_REPORT_FILE", &path);
+
+        #[cfg(unix)]
+        {
+            use std::os::unix::process::CommandExt;
+
+            let error = process::Command::new(exe_path).args(args).exec();
+            eprintln!("Failed to restart Luminol: {error:?}");
+            let _ = fs::remove_file(&path);
+        }
+        #[cfg(not(unix))]
+        {
+            if let Err(error) = std::process::Command::new(exe_path).args(args).spawn() {
+                eprintln!("Failed to restart Luminol: {error:?}");
+                let _ = std::fs::remove_file(&path);
+            }
+        }
+    }));
+
+    Ok(())
+}
+
+fn init_log() -> (sync::Arc<OnceCell<egui::Context>>, sync::mpsc::Receiver<u8>) {
+    let (log_byte_tx, log_byte_rx) = sync::mpsc::channel();
+    let ctx_cell = sync::Arc::new(OnceCell::new());
+    log::initialize_log(log_byte_tx, ctx_cell.clone());
+    (ctx_cell, log_byte_rx)
+}
+
+fn run_app(
+    report: Option<String>,
+    ctx_cell: sync::Arc<OnceCell<egui::Context>>,
+    log_byte_rx: sync::mpsc::Receiver<u8>,
+    #[cfg(feature = "steamworks")] steamworks: crate::steam::Steamworks,
+) -> Result<()> {
+    let icon_image = image::load_from_memory(ICON)?;
+
+    luminol_eframe::run_native(
+        "Luminol",
+        luminol_eframe::NativeOptions {
+            viewport: egui::ViewportBuilder::default()
+                .with_drag_and_drop(true)
+                .with_icon(egui::IconData {
+                    width: icon_image.width(),
+                    height: icon_image.height(),
+                    rgba: icon_image.to_rgba8().to_vec(),
+                })
+                .with_app_id("astrabit.luminol"),
+            wgpu_options: luminol_egui_wgpu::WgpuConfiguration {
+                supported_backends: wgpu::util::backend_bits_from_env()
+                    .unwrap_or(wgpu::Backends::PRIMARY | wgpu::Backends::SECONDARY),
+                // TODO: Load this value from a settings file
+                power_preference: wgpu::util::power_preference_from_env()
+                    .unwrap_or(wgpu::PowerPreference::LowPower),
+                ..Default::default()
+            },
+            persist_window: true,
+
+            ..Default::default()
+        },
+        Box::new(move |cc| {
+            unwrap_result(
+                ctx_cell
+                    .set(cc.egui_ctx.clone())
+                    .map_err(|_| Error::EguiContextCellAlreadySet),
+            );
+
+            Ok(Box::new(app::App::new(
+                cc,
+                report,
+                Default::default(),
+                log_byte_rx,
+                std::env::args_os().nth(1),
+                #[cfg(feature = "steamworks")]
+                steamworks,
+            )))
+        }),
+    );
```

Comment:
Can you re-add the `.expect("failed to start luminol")` here to remove the warning when compiling the native build?